### ISSUE TITLE
Virtual repeat keeps the scope values of the 'cloned' element

### ISSUE THREAD

**Issue by leseulsteve** [angular/material#8336]
I have list items that change state when they are clicked.

The problem is that state is then 'copied' in the next list items rendered below.

[code pen example](http://codepen.io/anon/pen/mPaRMW)

Is that the normal behavior?  Would it be better to 'copy' that element in is original state?

**Comment by RolandHeath** [comment]
This is a consequence of md-virtual-repeat re-using scopes, and not rendering everything at once, which is where the performance gain comes from.
If you need to apply state to a (virtual) list item, you need to apply it to the model for that item, not the view - with a virtual repeat, the view will cease to exist as soon as it goes too far out of sight.

I've modified your [codepen](http://codepen.io/RolandHeath/pen/jqXLXP) to an example of how to do this.

In short - This isn't exactly normal, but doing it any differently would lose the performance benefits of using a virtual repeat. So you're better off binding state to the underlying data, or (on a small data set) not using a virtual repeat.

**Comment by leseulsteve** [comment]
Thank you @RolandHeath for the response, that is actually what I wanted to try this morning. I was using angular-vs-repeat before and never ran into this issue with that lib.
